ginga/misc/plugins/LineProfile.py

```diff
@@ -0,0 +1,186 @@
+        ax = self.plot.add_axis()
+        ax.grid(False)
+
+        w = Widgets.wrap(self.plot.get_widget())
+        vbox.add_widget(w, stretch=1)
+
+        fr = Widgets.Frame("Axes controls")
+        self.hbox_axes = Widgets.HBox()
+        self.hbox_axes.set_border_width(4)
+        self.hbox_axes.set_spacing(1)
+        fr.set_widget(self.hbox_axes)
+
+        vbox.add_widget(fr, stretch=0)
+        self.build_axes()
+
+        # scroll bars will allow lots of content to be accessed
+        top.add_widget(sw, stretch=1)
+
+        # A button box that is always visible at the bottom
+        btns = Widgets.HBox()
+        btns.set_spacing(3)
+
+        # Add a close button for the convenience of the user
+        btn = Widgets.Button("Close")
+        btn.add_callback('activated', lambda w: self.close())
+        btns.add_widget(btn, stretch=0)
+        btns.add_widget(Widgets.Label(''), stretch=1)
+        top.add_widget(btns, stretch=0)
+
+        # Add our GUI to the container
+        container.add_widget(top, stretch=1)
+        self.gui_up = True
+
+    def build_axes(self):
+        self.hbox_axes.remove_all()
+        image = self.fitsimage.get_image()
+        if image is not None:
+            self.axes_states = []
+            # For easier mapping of indices with the axes
+            self.axes_states.append(None)
+
+            # Add Checkbox widgets
+            for i in xrange(1, len(image.get_mddata().shape)+1):
+                name = 'NAXIS%d' % i
+                chkbox = Widgets.CheckBox(name)
+                self.axes_states.append(False)
+                self.hbox_axes.add_widget(chkbox)
+
+                # Disable axes 1,2
+                if i < 3:
+                    chkbox.set_enabled(False)
+                    continue
+
+                # Add callback
+                self.axes_callback_handler(chkbox, i)
+
+    def axes_callback_handler(self, chkbox, pos):
+        chkbox.add_callback('activated', lambda w, tf: self.axis_toggle_cb(w, tf, pos))
+
+    def instructions(self):
+        self.tw.set_text("""Use the mouse to pick a point""")
+
+    def close(self):
+        chname = self.fv.get_channelName(self.fitsimage)
+        self.fv.stop_local_plugin(chname, str(self))
+        self.gui_up = False
+        return True
+        
+    def start(self):
+        self.instructions()
+
+        self.resume()
+
+    def pause(self):
+        pass
+
+    def resume(self):
+        self.redo()
+        
+    def stop(self):
+        pass
+        
+    def redo(self):
+        self.build_axes()
+
+        # Get handle to MultiDim plugin for this channel
+        chname = self.fv.get_channelName(self.fitsimage)
+        chinfo = self.fv.get_channelInfo(chname)
+        self.mddim = chinfo.opmon.getPlugin('MultiDim')
+
+        if not self.mddim.gui_up:
+            self.fv.start_local_plugin(chname, str('multidim'), None)
+
+        # Get image being shown
+        self.image = self.mddim.image
+
```

Comment:
Yeah, not necessary for accessing the header.  If it's not needed I would remove all that.